### ISSUE TITLE
Have an easy way to filter to just errors in logs view in dagit

### ISSUE THREAD

**Issue by dpeng817** [dagster-io/dagster#12741]
Right now, when viewing logs in dagit, by default all logs are shown for the run. There are toggles for every type of event that is displayed:
![Screen Shot 2023-03-06 at 9 35 06 AM](https://user-images.githubusercontent.com/17576880/223187393-85a883d5-b2db-4930-8b29-f6ece4b477df.png)
Often when debugging, you want to drill down to specifically the failure events. But in order to do that, you have to switch off the other types of events. The toggle system is actually very counterintuitive to how most people would expect to be filtering events in a debugging scenario. 

I am suggesting adding a shortcut button that appears when there are failure logs, that will automatically filter down to just failure logs.

**Comment by markfickett** [comment]
I agree; for my use case, one button to show run failures + event logs would be the ideal. Especially if you could jump right to that view from the failed run's listing in the job runs tab!

**Comment by hellendag** [comment]
It seems like we are also not persisting filter settings due to the URL querystring update. I'll plan to:

- Move the switch-tags into a selector component, to consolidate some of this UI and make it a little easier to use
- Persist the filter settings

**Comment by markfickett** [comment]
@hellendag Thanks for the quick UI improvement!

I think @dpeng817 's and my workflow remains a bit cumbersome though:

> I am suggesting adding a shortcut button that appears when there are failure logs, that will automatically filter down to just failure logs.

I have the same experience. When I find a failed run, I want to jump right to whatever logs will show me the failure. Even after your change, it's several clicks: turn off info, turn off event, turn on error. Persisting the filters won't solve that case because I do often want to see all the info/event logs, so I won't just leave it to set to errors all the time.

There's also this aspect of the ticket:

> Additionally, the run failure event is not shown when toggling to just error events which seems like an oversight.

The current available toggles don't make it easy to find `type:RUN_FAILURE`, which can sometimes include the critical error and is not surfaced by the error-level toggle.
